app.js

```diff
@@ -11,41 +11,90 @@ const taskManager = () => {
   function renderTodos() {
     $todoList.innerHTML = "";
 
-    todos.forEach((todo) => {
-      const todoElement = createTodo(todo);
+    todos.forEach((todo, index) => {
+      const todoElement = createTodo(todo, index);
       $todoList.appendChild(todoElement);
     });
   }
 
-  function createTodo(todo) {
+  function createTodo(todo, index) {
     const $li = document.createElement("li");
     const $todoText = document.createElement("span");
     $todoText.textContent = todo.title;
 
+    const $editText = document.createElement("input");
+    $editText.value = todo.title;
+    $editText.style.display = "none";
+
+    const $errorMsg = document.createElement("div");
+    $errorMsg.style.display = "none";
+
+    const editButton = createButton("Edit", () =>
+      onEditTodo($errorMsg, $todoText, $editText, index)
+    );
+
     $li.appendChild($todoText);
+    $li.appendChild($editText);
+    $li.appendChild(editButton);
+    $li.appendChild($errorMsg);
     return $li;
   }
 
+  const createButton = (text, onClick) => {
+    const $button = document.createElement("button");
+    
+    $button.innerText = text;
+    $button.onclick = onClick;
+    return $button;
+  };
+
   function onAddTodo() {
     const inputText = $todoInput.value.trim();
 
     if (inputText == "") {
-      showFeedback($feedbackMessage, "Please add todo text!");
+      showFeedback($feedbackMessage, "Please Enter a valid Todo item");
       return;
     }
-
     clearFeedback($feedbackMessage);
-
     const todo = {
       title: $todoInput.value,
       createdAt: Date.now(),
+      isEditing: false,
     };
     todos.push(todo);
 
     renderTodos();
 
     $todoInput.value = "";
   }
+  function onEditTodo($errorMsg, $todoText, $editText, index) {
+    todos[index].isEditing = !todos[index].isEditing;
+
+    if (todos[index].isEditing) {
+      $editText.value = todos[index].title;
+      $editText.style.display = "inline";
+      $todoText.style.display = "none";
+
+      $editText.focus();
+    } else {
+      const updateText = $editText.value.trim();
+
+      if (updateText) {
+        clearFeedback($errorMsg);
+
+        $editText.style.display = "none";
+        $todoText.style.display = "inline";
+
+        todos[index].title = updateText;
+        $todoText.textContent = updateText;
+      } else {
+        showFeedback($errorMsg, "Write something!");
+
+        todos[index].isEditing = true;
+        $editText.focus();
+      }
+    }
+  }
```

Comment:
https://github.com/hhe-wtag/dom-todo/pull/2/files#r1798148420

Here's a suggestion from my end to simplify todo editing. Let me know what do you think